src/go/rpk/pkg/cli/transform/init.go

```diff
@@ -0,0 +1,257 @@
+
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/cli/transform/buildpack"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/cli/transform/project"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/cli/transform/template"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/out"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+)
+
+type optionalBool struct {
+	ok *bool
+}
+
+func (b *optionalBool) Set(s string) error {
+	if strings.ToLower(s) == "prompt" {
+		b.ok = nil
+		return nil
+	}
+	v, err := strconv.ParseBool(s)
+	b.ok = &v
+	return err
+}
+
+func (*optionalBool) Type() string {
+	return "bool"
+}
+
+func (b *optionalBool) String() string {
+	if b.ok == nil {
+		return "prompt"
+	}
+	return strconv.FormatBool(*b.ok)
+}
+
+func (*optionalBool) IsBoolFlag() bool { return true }
+
+func newInitializeCommand(fs afero.Fs) *cobra.Command {
+	var (
+		name    string
+		lang    project.WasmLang
+		install optionalBool
+	)
+	cmd := &cobra.Command{
+		Use:   "init [DIRECTORY]",
+		Short: "Initialize a transform",
+		Long: `Initialize a transform.
+
+Creates a new transform using a template in the current directory.
+
+A new directory can be created by specifying it in the command, like:
+
+rpk transform init foobar
+
+Will initialize a transform project in the foobar directory.
+		`,
+		Args: cobra.MaximumNArgs(1),
+		Run: func(cmd *cobra.Command, args []string) {
+			path, err := computeProjectDirectory(fs, args)
+			out.MaybeDie(err, "unable to determine project directory: %v", err)
+			c := filepath.Join(path, project.ConfigFileName)
+			ok, err := afero.Exists(fs, c)
+			out.MaybeDie(err, "unable to determine if %q exists: %v", c, err)
+			if ok {
+				out.Die("there is already a transform at %q, please delete it before retrying", c)
+			}
+			for name == "" {
+				suggestion := filepath.Base(path)
+				if suggestion == "." {
+					suggestion = ""
+				}
+				name, err = out.PromptWithSuggestion(suggestion, "name this transform:")
+				out.MaybeDie(err, "unable to determine project name: %v", err)
+			}
+			if lang == "" {
+				langVal, err := out.Pick(project.AllWasmLangs, "select a language:")
+				out.MaybeDie(err, "unable to determine transform language: %v", err)
+				lang = project.WasmLang(langVal)
+			}
+			p := transformProject{Name: name, Lang: lang, Path: path}
+
+			fmt.Printf("generating project in %s...\n", p.Path)
+			err = executeGenerate(fs, p)
+			out.MaybeDie(err, "unable to generate all manifest files: %v", err)
+			fmt.Println("created project in", p.Path)
+
+			if install.ok == nil {
+				ok, err = out.Confirm("install dependencies?")
+				install.ok = &ok
+				out.MaybeDie(err, "unable to determine if dependencies should be installed: %v", err)
+			} else if *install.ok {
+				fmt.Println("installing dependencies...")
+			}
+			if *install.ok {
+				err = installDeps(cmd.Context(), fs, p)
+				out.MaybeDie(err, "unable to install dependencies: %v", err)
+				fmt.Println("dependencies installed")
+			}
```

Comment:
Can we merge these two ifs ?

Reply:
We could. There is a different if you are prompted we'll still write `installing dependencies...`, my intention was that if we prompted the user we can't skip that bit. Maybe it's not worth it... I'll merge them.